lib/buffer.js

```diff
@@ -146,9 +146,9 @@ function Buffer(arg, encodingOrOffset, length) {
   // Common case.
   if (typeof arg === 'number') {
     if (typeof encodingOrOffset === 'string') {
-      throw new Error(
-        'If encoding is specified then the first argument must be a string'
-      );
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', {
+        msg: 'The first argument must be of type string.'
```

Comment:
Agreed.
I think `invalidArgType(name, expected, actual)` is unsuitable for all situations. Because "Invalid argument type" can caused by many reasons:
- Argument must be one or more specified type, but recieve other.
- Argument must NOT be one or more specified type, but recieve.
- In some specified condition, the above two happen.

`invalidArgType(name, expected, actual)` in internal/errors.js can only handle the first one, without concerning about the rest.

Should I extend the `invalidArgType` method in order to make it suitable for all situations?